plugins/engines/ospray/ispc/render/SimulationRenderer.ispc

```diff
@@ -228,186 +229,157 @@ inline varying bool intersectBox(const uniform SimulationRenderer* uniform self,
-            //                    const varying vec2f s =
-            //                        make_vec2f(1.f / self->randomNumber);
-            //                    DifferentialGeometry dg;
-            //                    dg.P = point;
-            //                    const varying Light_SampleRes lightSample =
-            //                        light->sample(light, dg, s);
-
-            //                    Ray lightRay = ray;
-            //                    if (self->softShadows > 0.f)
-            //                        lightRay.dir = normalize(
-            //                            lightSample.dir +
-            //                            self->softShadows *
-            //                                getRandomVector(sample,
-            //                                lightSample.dir,
-            //                                                self->randomNumber));
-            //                    else
-            //                        lightRay.dir = lightSample.dir;
-
-            //                    lightRay.t =
-            //                    length(make_vec3f(self->volumeDimensions));
-            //                    lightRay.t0 = dg.epsilon;
-            //                    lightRay.org = point;
-            //                    lightRay.dir = lightSample.dir;
-
-            //                    shadowIntensity +=
-            //                        getVolumeShadowContribution(self,
-            //                        lightRay, sample);
-            //                }
-            //                const float giAttenuation =
-            //                    1.f - shadowIntensity * self->shadows;
-            //                voxelColor.x = voxelColor.x * giAttenuation;
-            //                voxelColor.y = voxelColor.y * giAttenuation;
-            //                voxelColor.z = voxelColor.z * giAttenuation;
-            //            }
-
-            // Compose final voxel color
-            vec4f voxelColor =
-                clamp(sampleOpacity / volume->samplingRate) *
-                make_vec4f(sampleColor.x, sampleColor.y, sampleColor.z, 1.0f);
-            composite(voxelColor, pathColor, (float)self->volumeSamplesPerRay);
+            // Voxel shading
+            if (self->shadingEnabled)
+            {
+                vec3f gradient =
+                    safe_normalize(volume->computeGradient(volume, point));
+                // Setup differential geometry for the volume sample point.
+                DifferentialGeometry dg;
+                dg.P = point;
+                dg.epsilon = epsilon;
+                dg.Ns = gradient;
+
+                gradient =
+                    safe_normalize(volume->computeGradient(volume, point));
+
+                // Shading according to computed normal
+                const vec2f s =
+                    make_vec2f(0.0f); // sample center of area lights
+                const Light_SampleRes light =
```

Comment:
Since this is only taking the first light what happens if you have several lights?